src/main/webapp/app/overview/exercise-details/header-exercise-page-with-details.component.ts

```diff
@@ -0,0 +1,36 @@
+import { Component, Input, OnInit, OnChanges, SimpleChanges } from '@angular/core';
+import * as moment from 'moment';
+import { Exercise, ExerciseCategory, ExerciseService, getIcon } from 'app/entities/exercise';
+
+@Component({
+    selector: 'jhi-header-exercise-page-with-details',
+    templateUrl: './header-exercise-page-with-details.component.html',
+})
+export class HeaderExercisePageWithDetailsComponent implements OnInit, OnChanges {
+    @Input() public exercise: Exercise;
+    @Input() public onBackClick: () => void;
+    @Input() public title: string;
+
+    public exerciseStatusBadge = 'badge-success';
+    public exerciseCategories: ExerciseCategory[];
+
+    getIcon = getIcon;
+
+    constructor(private exerciseService: ExerciseService) {}
+
+    ngOnInit(): void {
+        this.setExerciseStatusBadge();
+        this.exerciseCategories = this.exerciseService.convertExerciseCategoriesFromServer(this.exercise);
+    }
+
+    ngOnChanges(changes: SimpleChanges): void {
+        this.setExerciseStatusBadge();
+        this.exerciseCategories = this.exerciseService.convertExerciseCategoriesFromServer(this.exercise);
+    }
+
+    private setExerciseStatusBadge(): void {
```

Comment:
i think we can remove the function from the `src/main/webapp/app/overview/exercise-details/course-exercise-details.component.ts` file

Reply:
Right, fixed